src/services/shareService.ts

```diff
@@ -0,0 +1,184 @@
+'use server';
+import connectDB from '../lib/dbConnect';
+import { stringToObjectId } from '../lib/utils';
+
+import CardAdaptor from './CardAdaptor';
+import { createCard } from './cardService';
+import { getTypes } from './typeService';
+
+import { ICard } from '@/models/Card';
+import SharingModel, { ISharing } from '@/models/Sharing';
+
+export async function getSharings() {
+    try {
+        await connectDB();
+
+        // TODO: Get USER_ID from session
+        const user = stringToObjectId('123456789012');
+
+        if (!user) {
+            return { error: 'user not found' };
+        }
+
+        const sharings = await SharingModel.find<ISharing>({ user });
+
+        return {
+            sharings,
+        };
+    } catch (error) {
+        return { error };
+    }
+}
+
+export async function createSharing(cardId: string, validHours = (24*7)) {
+    try {
+        await connectDB();
+        
+        const card = stringToObjectId(cardId);
```

Comment:
I think the variable `card` is objectId instead of Card object.

Reply:
Fixed: https://github.com/tsoibet/kaado/pull/14/commits/425ff6b2bd5601e42d11f7ef8697111da4cd020f